core/zoom_controls.js

```diff
@@ -53,6 +53,35 @@ Blockly.ZoomControls = function(workspace) {
    * @private
    */
   this.onZoomOutWrapper_ = null;
+
+  /**
+   * The vertical distance between the workspace bottom edge and the control.
+   * The value is initialized during `init`.
+   * @type {?number}
+   * @private
+   */
+  this.verticalSpacing_ = null;
+
+  /**
+   * The zoom in svg <g> element.
+   * @type {SVGGElement}
```

Comment:
These should have explicit `?` if they are nullable, I think.

Reply:
Generally the rule is to not use redundant types. In this case, SVGGElement is an object, and objects are implicitly nullable, so no need to add the `?`. 
Literals on the other hand (string, boolean, number) are non-nullable implicitly. I know it's all a little confusing!

Reply:
According to this, reference types should always be annotated. https://google.github.io/styleguide/jsguide.html#jsdoc-nullability
I think this changed at some point, so maybe Blockly is using an older version of the style guide? Elsewhere in this file some classes are annotated with `?` though

Reply:
Yeah I think this has changed over time.

Elsewhere in the file, I think you're referring to `Blockly.EventData`. That is not an Object, but a record type, and they are non-nullable by default.
This is a case in point that what we are doing in Blockly is confusing. We really _should_ be doing what the style guide suggests there, and always add the modified for anything that's not a literal to be clear about its nullability.

I think that's going to be a heavy lift to switch over. Let's discuss at the meeting next week, and see if we should look into doing this as a big PR (with some find and replace), and switch over the Blockly style to match the Google style.

For this PR, I'll keep as is.

Reply:
Sounds good!